### PR TITLE
Enable AdamW optimizer 

### LINKED ISSUE #6153: Enable AdamW Optimizer
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

When we use Adam, we have to tune learning rate along with the batch size.
It is cumbersome; with [AdamW](https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html), we don't have to re-tune learning rate even if we change batch size.
So, it is nice to be able to use this option.

I have created PR to enable AdamW optimizer. Please check it out.
#6152

### Use case

_No response_

### Additional

_No response_

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

### PR DESCRIPTION
**Pull request by bilzard**
This PR will allow you to use AdamW for the optimizer.

## Background

 #6153

## Backward Compatibility Changes

The `--adam` option will be deprecated.

## Operational Verification

- [x] `--optimizer Adam` -> `Adam` is chosen
- [x] `--optimizer AdamW` -> `AdamW` is chosen
- [x]  `--optimizer SGD` -> `SGD` is chosen
- [x]  no `--optimizer` option -> `SGD` is chosen
- [x] `--optimizer Hoge` -> `train.py: error: argument --optimizer: invalid choice: 'Hoge' (choose from 'SGD', 'Adam', 'AdamW')`

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
This PR introduces AdamW optimizer support to the YOLOv5 training script.

### 📊 Key Changes
- Added `AdamW` to the list of available optimizers in the `train.py`.
- Replaced the `--adam` argument in the argument parser to a more general `--optimizer` which accepts `'SGD'`, `'Adam'`, and `'AdamW'` as choices.
- Conditionally initialize the optimizer based on the new `--optimizer` argument, allowing for selection between `SGD`, `Adam`, and `AdamW`.

### 🎯 Purpose & Impact
- **Purpose**: To offer an alternative optimization algorithm that may improve training performance for specific scenarios or datasets.
- **Impact**: Users will now have the flexibility to choose `AdamW` during training, which can potentially lead to better convergence due to its different approach in handling weight decay compared to traditional `Adam`.

### CODE REVIEW

**Comment by glenn-jocher** [comment]
@bilzard PR is merged. Thank you for your contributions to YOLOv5 🚀 and Vision AI ⭐

**Comment by glenn-jocher** [comment]
@bilzard SGD/Adam batch size study results are here: https://wandb.ai/glenn-jocher/study-Adam. Dividing the SGD LR by 10 manually for the Adam/AdamW runs.

```python
# VOC
for b, m in zip([16, 64, 16, 64, 16, 64], ['SGD', 'SGD', 'Adam', 'Adam', 'AdamW', 'AdamW']):  # zip(batch_size, model)
  hyp = 'hyp.finetune.yaml' if m.startswith('SGD') else 'hyp.finetuneAdam.yaml'
  !python train.py --batch {b} --weights yolov5s.pt --data VOC.yaml --epochs 50 --cache --img 512 --nosave --hyp {hyp} --project study-Adam --name {m}-{b} --optimizer {m}
```

Adam seems to handle batch-size changes without issue, so it seems like no changes are required:
Screen Shot 2022-01-03 at 9 39 29 AM

**Comment by bilzard** [comment]
@glenn-jocher Thank you. It will be really helpful.

**Comment by Robotatron** [comment]
Does Adam or AdamW bring any noticeable differences from SGD in YOLO5 ?
Which one would you recommend for better mAP scores?
